h/search/query.py

```diff
@@ -94,8 +94,16 @@ def extract_limit(params):
 def extract_sort(params):
     return [{
         params.pop("sort", "updated"): {
-            "ignore_unmapped": True,
             "order": params.pop("order", "desc"),
+
+            # `unmapped_type` causes unknown fields specified as arguments to
+            # `sort` behave as if all documents contained empty values of the
+            # given type. Without this, specifying eg. `sort=foobar` throws
+            # an exception.
+            #
+            # We use the field type `boolean` to assist with migration because
+            # that exists in both ES 1 and ES 6.
+            "unmapped_type": "boolean",
```

Comment:
The documentation <https://www.elastic.co/guide/en/elasticsearch/reference/1.5/search-request-sort.html> hints that the intention is to be able to set the type of the missing field value such that it sorts in the expected order. I'm a little wary of setting this blanket setting to boolean. It seems they did that here <https://github.com/elastic/kibana/issues/4593> and they ended up having to change it to the proper type so that results showed up in the proper order. It's not exactly clear from the docs how this behaves in practice so I think a test is probably required here to see how this works. It's also possible we already have a test and I'm not aware of it.

Reply:
es 1.5 types: https://www.elastic.co/guide/en/elasticsearch/reference/1.5/mapping-core-types.html
es 6.2 types: https://www.elastic.co/guide/en/elasticsearch/reference/6.2/mapping-types.html

It seems like the most used type we have for fields is probably a string/text/keyword. Note that is a breaking change between es versions that we already have so I wouldn't seek to avoid it as we will be forced to address it regardless. The real question here tho is what **missing** field's are there and what are their types.

Reply:
The purpose of `ignore_unmapped` / `unmapped_type` here is to avoid throwing an exception if the user tries to sort by an unknown field - eg. `/api/search?sort=foobar`. My understanding is that the type can matter if you are trying to execute a search across multiple indexes ("cross index search") and the field exists in some indexes' mappings but not others (see [this motivating issue](https://github.com/elastic/elasticsearch/issues/808)). In our case we only have a single active index and we are not interested in doing any meaningful sorting if the field name is unknown, so I don't think the type matters.

The existing behaviour if a search API query specifies an unknown field to sort by is that the query succeeds but results come back unsorted. Whether that is the ideal behaviour for the API is debatable, but I didn't want to change it here.

I can add a test to clarify this.